### PR TITLE
Route metric information

### PR DESCRIPTION
**Pull request by alirezaDavid**
* Fixed a typo
* Add route metric information 

```
openshift_route_info{route="route1",namespace="ns1",host="example.com",path="",tls_termination="edge",to_kind="Service",to_name="svc1",to_weight="100"} 1
openshift_route_status{route="route1",namespace="ns1",host="example.com",status="True",type="Admitted",router_name="router1"} 1
```

### CODE REVIEW

File: pkg/collectors/route.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+					f.Metrics = append(f.Metrics, &metric.Metric{
+						Value: float64(d.CreationTimestamp.Unix()),
+					})
+				}
+
+				return f
+			}),
+		},
+		metric.FamilyGenerator{
+			Name: "openshift_route_info",
```
Comment (wanghaoran1988) [inline pkg/collectors/route.go]: Could you split this into to metrics ? "openshift_route_info" that includes the info in the "spec", and "openshift_route_status" that describe the "status" part? WTDY?

**Comment by alirezaDavid** [comment]
> Could you also add unit test?

Done 🙂

**Comment by alirezaDavid** [comment]
@wanghaoran1988 Ping :)

File: pkg/collectors/route.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kube-state-metrics/pkg/metric"
+	"k8s.io/kube-state-metrics/pkg/version"
+	"strconv"
+)
```
Comment (wanghaoran1988) [inline pkg/collectors/route.go]: Could you sorted the dependencies like this? split the golang libary, github, openshift and k8s.io dependencies?
```
	"strconv"

	"k8s.io/apimachinery/pkg/runtime"
	"k8s.io/apimachinery/pkg/watch"
	"k8s.io/client-go/tools/cache"
	"k8s.io/client-go/tools/clientcmd"
	"k8s.io/kube-state-metrics/pkg/metric"
	"k8s.io/kube-state-metrics/pkg/version"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/golang/glog"

	"github.com/openshift/api/route/v1"
	routeclient "github.com/openshift/client-go/route/clientset/versioned"
```

**Comment by wanghaoran1988** [comment]
Generally lgtm, @aditya-konarde Could you take a look also ?

**Comment by aditya-konarde** [comment]
Sorry for being off. At Kubecon right now. but thanks for the very comprehensive review @wanghaoran1988

**Pull request merged**